das/celestia/celestia.go

```diff
@@ -380,14 +378,12 @@ func (c *CelestiaDA) Read(ctx context.Context, blobPointer *types.BlobPointer) (
 	headerDataHash := [32]byte{}
 	copy(headerDataHash[:], header.DataHash)
 	if headerDataHash != blobPointer.DataRoot {
-		log.Error("Data Root mismatch", " header.DataHash", header.DataHash, "blobPointer.DataRoot", hex.EncodeToString(blobPointer.DataRoot[:]))
```

Comment:
this has to explicitly return an error, otherwise it would allow the batch poster to post an invalid batch and the nodes reading it could accept it. I guess for now we should just keep this as a dangerous config that only the chain operator uses, and other nodes keep off by default.